Port to Qt6

lkorbel opened the pull request:
This version works with Qt6.
Also few more data types added.

Commit:
Add QUrl type to MConfig

Commit:
Fix bug in mconfig-crypto option

Previously, if mcrypto was added to the project, mconfig will use it regardless
of related cmake option value which may lead to runtime crashes (and did in my case).

Commit:
Port code to Qt6

There are few depracated methods in QVariant. Code remain back compatible with Qt5.
Added CMakeLists for Qt6.

Commit:
add support for uint type

Commit:
update cmake config to work with bot Qt6 and Qt5

Linking against versionless Qt libraries is now recomended.

lkorbel commented:
@sierdzio Hi Tom! Sending my current modifications to the module. Do you plan to continue public releases for this one?

sierdzio commented:
> @sierdzio Hi Tom! Sending my current modifications to the module. Do you plan to continue public releases for this one?

Yes, when we will do any changes, they will be pushed here also. But since we're mostly busy with external projects now, nobody is paying much attention to MCDB stuff :-(

sierdzio commented on CMakeLists.txt:
```diff
@@ -42,21 +45,20 @@ target_compile_definitions(mconfig PUBLIC MCONFIG_LIB)
 option(mconfig-crypto "MConfig: Build with encryption functions" OFF)
 if(mconfig-crypto)
+  message("MConfig: using encryption library")
+  target_compile_definitions(mconfig PRIVATE CRYPTED_CONFIG)
```
@lkorbel should probably be `ENCRYPTED_CONFIG`.

Also, similar changes will be required for qmake support (.pri file)

lkorbel replied:
@sierdzio Thank you, I will rename it.

Commit:
fix build

Commit:
rename CRYPTED_CONFIG to ENCRYPTED_CONFIG

Commit:
update configuration for encrypted config

QMake support is limited to defining proper macro. Readme file contain updated instructions

Commit:
refactor unit test

MConfig has very limited functionality that is saving/loading value of specific types
that can be modified during runtime. Therefore data driven test its not very suitable
here. We should however test this basic scenario for all supported types. New types can be added
by simply calling template functions. There are 3 versions of test_save_and_load template that
relate to 3 common usecases for MConfig:
1. Rely on default QSettings by calling MConfig::load()/save()
2. Enforce use of Ini file by providing its path to methods above.
3. Use encrypted config by enabling proper build options and providing passphrase

Commit:
deduct type of variable in CONFIG_VALUE macro

User does not have to provide MetaType id for variable as it can be easily deducted at compile time.
This change makes configuration slightly easier and more robust since chance of type mismatch is eliminated.
Although this change breaks compatibility with previous versions update is very simple - for old lines like:
CONFIG_VALUE(my_string, QMetaType::QString)
just remove 2nd parameter and leave variable only:
CONFIG_VALUE(my_string)

Commit:
fix examples

Commit:
update documentation

sierdzio commented:
OKi let's merge. I guess we'll just abandon our GitLab copy and move development here, if it gets restarted
